### PR TITLE
MAINT/TST: pytorch-ify torch._numpy tests (added tests only, not vendored)

### PR DESCRIPTION
**Pull request by ev-br**
1. Inherit from TestCase
2. Use pytorch parametrization
3. Use unittest.expectedFailure to mark xfails

All this to make pytest-less invocation work:

$ python test/torch_np/test_basic.py

Furthermor, tests can now be run under dynamo, and we see first errors:

```
$ PYTORCH_TEST_WITH_DYNAMO=1 python test/torch_np/test_basic.py -k test_toscalar_list_func
.E.
======================================================================
ERROR: test_toscalar_list_func_<function shape at 0x7f9b83a4fc10>_np_func_<function shape at 0x7f9a8dd38af0> (__main__.TestOneArrToScalar)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ev-br/repos/pytorch/torch/testing/_internal/common_utils.py", line 356, in instantiated_test
    test(self, **param_kwargs)
  File "test/torch_np/test_basic.py", line 232, in test_toscalar_list
    @parametrize("func, np_func", one_arg_scalar_funcs)
  File "/home/ev-br/repos/pytorch/torch/nn/modules/module.py", line 1519, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ev-br/repos/pytorch/torch/nn/modules/module.py", line 1528, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ev-br/repos/pytorch/torch/_dynamo/eval_frame.py", line 406, in _fn
    return fn(*args, **kwargs)
  File "/home/ev-br/repos/pytorch/torch/fx/graph_module.py", line 726, in call_wrapped
    return self._wrapped_call(self, *args, **kwargs)
  File "/home/ev-br/repos/pytorch/torch/fx/graph_module.py", line 305, in __call__
    raise e
  File "/home/ev-br/repos/pytorch/torch/fx/graph_module.py", line 292, in __call__
    return super(self.cls, obj).__call__(*args, **kwargs)  # type: ignore[misc]
  File "/home/ev-br/repos/pytorch/torch/nn/modules/module.py", line 1519, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ev-br/repos/pytorch/torch/nn/modules/module.py", line 1528, in _call_impl
    return forward_call(*args, **kwargs)
  File "<eval_with_key>.2", line 5, in forward
    shape = torch._numpy._funcs_impl.shape([[1, 2, 3], [4, 5, 6]])
  File "/home/ev-br/repos/pytorch/torch/_numpy/_funcs_impl.py", line 655, in shape
    return tuple(a.shape)
AttributeError: 'list' object has no attribute 'shape'

----------------------------------------------------------------------
Ran 3 tests in 0.915s

FAILED (errors=1)
```

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/torch_np/test_basic.py [hunk 1]
```diff
@@ -487,7 +489,7 @@ def test_divmod_out_list(self):
         assert quot is out[0]
         assert rem is out[1]
 
-    @pytest.mark.xfail(reason="out1, out2 not implemented")
+    @xfail  # ("out1, out2 not implemented")
```
Comment (lezcano) [inline test/torch_np/test_basic.py:492]: why not keeping the comment?

**Review by lezcano — APPROVED** [review]
Nice! I really think this can be automatised via a script though. The only change that's not local is moving the parametrize from the class to the methods. Rather than doing that, it may be simpler to implement support for parametrising classes in our `parametrize`.

File: test/torch_np/test_basic.py (hunk 1 above)
Comment (ev-br) [reply test/torch_np/test_basic.py:492]: It just does not seem to have a relevant argument.

```
In [1]: from unittest import expectedFailure

In [2]: expectedFailure?
Signature: expectedFailure(test_item)
Docstring: <no docstring>
File:      ~/.conda/envs/pytorch-dev/lib/python3.8/unittest/case.py
Type:      function
```

**Comment by ev-br** [comment]
@lezcano this PR is now clean under the checker from https://github.com/pytorch/pytorch/pull/109815 and the remaining CI failure is a matter of a more fine-grained xfail decorator (works locally, should work on CI, too)

File: test/torch_np/test_reductions.py [hunk 2]
```diff
@@ -259,7 +269,42 @@ def test_sum_where(self):
...
+    np.any,
+    np.all,
+    np.argmin,
+    np.argmax,
+    np.prod,
+}
+
+fails_out_arg = {
+    np.count_nonzero,
+}
```
Comment (lezcano) [inline test/torch_np/test_reductions.py:303]: this one we could probably fix?

File: test/torch_np/test_ndarray_methods.py [hunk 3]
```diff
@@ -179,16 +189,21 @@ class TestArgmaxArgminCommon:
...
+                *[
+                    [
+                        (size, axis)
+                        for axis in list(range(-len(size), len(size))) + [None]
+                    ]
+                    for size in sizes
+                ]
+            )
         ),
     )
```
Comment (lezcano) [inline test/torch_np/test_ndarray_methods.py:205]: thank you linter, what would we do without you...

**Review by lezcano — APPROVED** [review]

File: test/torch_np/test_reductions.py (hunk 2 above)
Comment (ev-br) [reply test/torch_np/test_reductions.py:303]: If we want to do a superset of numpy here, which does not have the out= arg :-) 

https://numpy.org/doc/stable/reference/generated/numpy.count_nonzero.html

Comment (lezcano) [reply test/torch_np/test_reductions.py:303]: oh, it doesn't. So this is expected then? It should probably be a skip then (no need to act on this)

**Pull request closed without merging**